cogs/Announce.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python
+
+"""
+Koala Bot Announce feature
+Created by: Bill Cao
+"""
+
+# Built-in/Generic Imports
+import asyncio
+
+# Libs
+import math
+from typing import Optional, Tuple
+
+import discord
+from discord.ext import commands
+from utils.KoalaUtils import extract_id
+from utils import KoalaColours
+import time
+
+# constants
+ANNOUNCE_SEPARATION_DAYS = 30
+SECONDS_IN_A_DAY = 86400
+TIMEOUT_TIME = 60.0
+MAX_MESSAGE_LENGTH = 2000
+
+import KoalaBot
+
+
+def announce_is_enabled(ctx):
+    """
+    A command used to check if the guild has enabled announce
+    e.g. @commands.check(announce_is_enabled)
+
+    :param ctx: The context of the message
+    :return: True if enabled or test, False otherwise
+    """
+    try:
+        result = KoalaBot.check_guild_has_ext(ctx, "Announce")
+    except PermissionError:
+        result = False
+
+    return result or (str(ctx.guild) == KoalaBot.TEST_USER and KoalaBot.is_dpytest)
+
+
+async def wait_for_message(bot: discord.Client, ctx: commands.Context, timeout: float = TIMEOUT_TIME) -> Tuple[
+    Optional[discord.Message], Optional[discord.TextChannel]]:
+    """
+        Wraps bot.wait_for with message event, checking that message author is the original context author. Has default
+        timeout of 60 seconds.
+        :param bot: Koala Bot client
+        :param ctx: Context of the original command
+        :param timeout: Time to wait before raising TimeoutError
+        :return: If a message (msg) was received, returns a tuple (msg, None). Else returns (None, ctx.channel)
+        """
+    try:
+        msg = await bot.wait_for('message', timeout=timeout, check=lambda message: message.author == ctx.author)
+    except (Exception, TypeError):
+        return None, ctx.channel
+    if not msg:
+        return msg, ctx.channel
+    return msg, None
```

Comment:
This is also used in ReactForRole (line 698), IntroCog (line 33), please can you move this to `/utils/KoalaUtils.py` and refactor this and the other codes to use the relocated method and then you can remove the duplicate `wait_for_message` methods

Reply:
@Copper76 has this been completed?

Reply:
I moved it to util and changed my cog, did you mean to let me mess with RFR and IntroCog as well, as I didn't want to destroy some working code by accident

Reply:
Yeah, moving those over to use the same method in utils would be great and reduce code redundancy

Reply:
k... time to change the test files

Reply:
The only problem is I don't quite understand some of the test cases and files and it may be easier for the authors to change them

Reply:
It should only require changing references to the method, everything else should still work

Reply:
Apparently not, since I changed it for RFR, which had the exact same code and half the tests failed

Reply:
Fixed RFR, IntroCog doesn't want to behave (and is slightly different). Resolved